CDLUC3/dmsp_frontend_prototype
# Auth-related updates to middleware and to error handler, refreshAuthTokens function and client components

## Description
Some auth-related updates were needed to the frontend to help it function properly.
- Updated middleware to redirect to `/login` if both access and refresh tokens were missing
- Added router as a param to `handleApolloErrors` function, so that it can redirect to `/login` 
- Updated template history page to use `handleApolloErrors` to handle errors from the graphql request
- Updated refreshAuthTokens to throw errors so that the error handler can then handle it accordingly.

Fixes # ([142](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/142))

## Type of change

## How Has This Been Tested?
Updated unit tests, and manually tested the `/templates/[templateId]/history` page and` /client-graphql-test` pages to make sure that access tokens were refreshed when the backend returned a 401 error.

- Run unit tests to make sure that they pass
- Manual test 1
-- Start the backend server
-- Start NextJS app, and log in and confirm that you can see both the `dmspt` and `dmspr` access and refresh token cookies
-- Navigate to `/template/1/history` page
-- Delete the `dmspt` cookie and refresh
-- Note the 401 error returned from the backend, and that the `/apollo-refresh` endpoint was called and that the `dmspt` access token cookie was added back

- Manual Test 2
-- Delete both the access and refresh token
-- Navigate to `/template/1/history` page
-- You should be redirected to the /login page

- Manual Test 3
-- Delete both the access and refresh token
-- Navigate to `/client-graphql-test
-- You should NOT be redirected to the /login page since this page is not protected


## Checklist:
- [NA] I have completed manual or automated accessibility testing for my changes
- [NA] Any dependent changes have been merged and published in downstream modules
